src/parser/core/modules/Interrupts.tsx

```diff
@@ -0,0 +1,143 @@
+import Suggestions, {SEVERITY, TieredSuggestion} from 'parser/core/modules/Suggestions'
+import Timeline from 'parser/core/modules/Timeline'
+import {Data} from './Data'
+
+interface SeverityTiers {
+	[key: number]: number
+}
+
+// used for timeline viewing by giving you a nice 30s window
+const TIMELINE_UPPER_MOD: number = 30000
+
+export abstract class Interrupts extends Module {
+	static handle: string = 'interrupts'
+	static title: MessageDescriptor = t('core.interrupts.title')`Interrupted Casts`
+
+	@dependency private timeline!: Timeline
+	@dependency private globalCooldown!: GlobalCooldown
+	@dependency private suggestions!: Suggestions
+	@dependency private data!: Data
+
+	private currentCast?: CastEvent | undefined
+	private droppedCasts?: CastEvent[] = []
+	private missedTimeMS: number = 0
+
+	/**
+	 * Implementing modules MAY override the icon to be used for the suggestion,
+	 * though, let's face it – interject is pretty much the perfect one.
+	 */
+	protected icon: string = ACTIONS.INTERJECT.icon
+
+	/**
+	 * Implementing modules MAY override the severity tiers for interrupted casts
+	 */
+	protected severity: SeverityTiers = {
+		2: SEVERITY.MEDIUM,
+		5: SEVERITY.MAJOR,
+	}
+
+	/**
+	 * Implementing modules MAY override the default suggestion text
+	 */
+	protected suggestionContent: JSX.Element | string = <Trans id="core.interrupts.suggestion.content">
+		Avoid interrupting casts by either prepositioning yourself or utilizing slidecasting where possible.
+	</Trans>
+
+	/**
+	 * Implementing modules MAY override this function to provide specific text if they wish for the 'why'
+	 * @param missedCasts
+	 * @param missedTime
+	 */
+	private suggestionWhy(missedCasts: CastEvent[], missedTime: number): JSX.Element | string {
+		return <Trans id="core.interrupts.suggestion.why">You missed { missedCasts.length } casts (approximately { this.parser.formatDuration(missedTime) } of total casting time) due to interruption.</Trans>
+	}
+
+	protected init() {
+		this.addEventHook('begincast', {by: 'player'}, this.onBeginCast)
+		this.addEventHook('cast', {by: 'player'}, this.onCast)
+
+		this.addEventHook('complete', this.onComplete)
+	}
+
+	private onBeginCast(event: CastEvent) {
+		// if they started casting something, then cancel and start something else,
+		// that's definitely an interrupted cast
+		if (this.currentCast) {
+			this.pushDropCasts(event)
+		}
+		this.currentCast = event
+	}
+
+	private onCast(event: CastEvent) {
+		const guid = event.ability.guid
+		// if the thing they started casting doesn't match up with what they cast, then
+		// that's an interrupted cast
+		if (this.currentCast && guid !== this.currentCast.ability.guid) {
+			this.pushDropCasts(event)
+		}
+		this.currentCast = undefined
+	}
+
+	private pushDropCasts(currentEvent: CastEvent) {
+		this.missedTimeMS += Math.min(this.globalCooldown.getEstimate(), currentEvent.timestamp - this.currentCast!.timestamp)
+		this.droppedCasts!.push(currentEvent)
```

Comment:
Since `droppedCasts` is initialised as an empty array, it'll always exist. When you remove the `?` in it's declaration, this ignore null check isn't required. Is using the GCD here a good idea though? I think using the current cast timestamp should be fine, although I'm not sure how safe ignoring it being null is.

Reply:
I think the reason why I used the estimated GCD here was because there was times where someone would interrupt a cast and then not cast for a while – for example, a BLM might not be able to do anything during E3S transition, which might mean you have a 30s 'gap' in your interrupted casts, and it seemed unfair to penalize them for more than the gcd estimate.

I'm very open to suggestions here, though.

Reply:
That explanation seems reasonable to me, the ABC module should handle if they're straight up not casting vs unable to cast.